### PR TITLE
1023 Address frame delta initial contact print files

### PR DESCRIPTION
**Pull request by AdamHawtin**
# Motivation and Context:

We can now create action rules using a String which should resemble the end of a WHERE clause in SQL, e.g. `treatment_code IN ('abc', 'def')`. 

# What has changed?

* Renamed classifier and changed type

# How to test?

See other related PRs

# Links:

https://trello.com/c/TnldEmYZ/1023-address-frame-delta-generate-additional-initial-contact-files-13

### CODE REVIEW

**Comment by AdamHawtin** [comment]
[Address Frame Delta - generate additional initial contact files (13)](https://trello.com/c/TnldEmYZ/1023-address-frame-delta-generate-additional-initial-contact-files-13)

**Review by NickGrantX — CHANGES_REQUESTED** [review]
ATs look good. This needs the change to reflect that "users" are not defining our classifiers

File: acceptance_tests/features/steps/sample.py [hunk 1]
```diff
@@ -11,27 +14,14 @@ def load_sample_file_step(context, sample_file_name):
...
     get_and_check_uac_updated_messages(context)
 
-    # TODO: in a nice world this would be well nicer
-    # However when we set up the skeleton cases we need to use the census action-plan-id, which is fixed so the query
-    # returns old matching rows too.
-    # So we then add in > start_of_test_datetime, excellent, except db is on UTC,
-    # sometimes different on local machine. So doesn't work
-    # could possibly get the DB start time at the start of the test, but a bit OTT
-    if check_loaded_into_db:
-        poll_until_sample_is_ingested_to_action(context)
```
Comment (AdamHawtin) [inline acceptance_tests/features/steps/sample.py:34]: `poll_until_sample_is_ingested_to_action` now filters by the test start time so this check can be run everywhere this step is used

File: acceptance_tests/utilities/action_helper.py [hunk 2]
```diff
@@ -5,12 +5,15 @@
...
+from acceptance_tests.utilities.mappings import CLASSIFIERS_FOR_ACTION_TYPE, DEFAULT_CLASSIFIERS
 from acceptance_tests.utilities.test_case_helper import test_helper
 from config import Config
 
 
-def poll_until_sample_is_ingested_to_action(context):
-    sql_query = """SELECT count(*) FROM actionv2.cases WHERE action_plan_id = %s"""
+def poll_until_sample_is_ingested_to_action(context, after_date_time=None):
+    if not after_date_time:
+        after_date_time = context.test_start_utc
```
Comment (AdamHawtin) [inline acceptance_tests/utilities/action_helper.py:15]: Unless a specific date time is supplied, filter this check to only cases created after this scenarios start time. This is of importance for the scenarios that have to reuse the same 'census' action plan ID as it ignores cases loaded against that ID from previous test runs.

File: acceptance_tests/utilities/action_helper.py [hunk 3]
```diff
@@ -5,12 +5,15 @@
...
+    query = "SELECT count(*) FROM actionv2.cases WHERE action_plan_id = %s AND created_date_time > %s"
```
Comment (AdamHawtin) [inline acceptance_tests/utilities/action_helper.py:16]: Unless a specific date time is supplied,  this check will default to filtering to only cases created after this scenarios start time. This is of importance for the scenarios that have to reuse the same 'census' action plan ID as it ignores cases loaded against that ID from previous test runs.

File: acceptance_tests/features/steps/sample.py [hunk 4]
```diff
@@ -43,3 +33,20 @@ def load_sample_file_successfully_step(context, sample_file_name, check_loaded_i
 def load_sample_file_and_check_qids(context, sample_file_name, questionnaire_types):
     load_sample_file_helper(context, sample_file_name)
     get_and_test_case_and_uac_msgs_are_correct(context, questionnaire_types)
+
+
+@step('{wait:d} seconds later delta sample file "{sample_file_name}" is loaded successfully')
+@step('{wait:d} second later delta sample file "{sample_file_name}" is loaded successfully')
```
Comment (NickGrantX) [inline acceptance_tests/features/steps/sample.py:38]: do we really need two steps for very subtly different grammatical pedantry?

**Review by NickGrantX — APPROVED** [review]

File: acceptance_tests/features/steps/sample.py (hunk 4 above)
Comment (LukeLoze1) [reply acceptance_tests/features/steps/sample.py:38]: The @step('{wait:d} seconds later delta sample file "{sample_file_name}" is loaded successfully') isn't used anywhere

**Review by LukeLoze1 — CHANGES_REQUESTED** [review]

**Review by LukeLoze1 — APPROVED** [review]

**Pull request merged**